examples/runners/wgpu/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+            }
+            Event::Suspended => {
+                surface = None;
+                swap_chain = None;
+            }
+            Event::WindowEvent {
+                event: WindowEvent::Resized(size),
+                ..
+            } => {
+                // Recreate the swap chain with the new size
+                sc_desc.width = size.width;
+                sc_desc.height = size.height;
+                if let Some(surface) = &surface {
+                    swap_chain = Some(device.create_swap_chain(surface, &sc_desc));
+                }
+            }
+            Event::RedrawRequested(_) => {
+                if let Some(swap_chain) = &mut swap_chain {
+                    let frame = swap_chain
+                        .get_current_frame()
+                        .expect("Failed to acquire next swap chain texture")
+                        .output;
+                    let mut encoder = device
+                        .create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
+                    {
+                        let mut rpass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+                            color_attachments: &[wgpu::RenderPassColorAttachmentDescriptor {
+                                attachment: &frame.view,
+                                resolve_target: None,
+                                ops: wgpu::Operations {
+                                    load: wgpu::LoadOp::Clear(wgpu::Color::GREEN),
+                                    store: true,
+                                },
+                            }],
+                            depth_stencil_attachment: None,
+                        });
+                        rpass.set_pipeline(&render_pipeline);
+                        rpass.draw(0..3, 0..1);
+                    }
+
+                    queue.submit(Some(encoder.finish()));
+                }
+            }
+            Event::WindowEvent {
+                event: WindowEvent::CloseRequested,
+                ..
+            } => *control_flow = ControlFlow::Exit,
+            Event::WindowEvent {
+                event:
+                    WindowEvent::KeyboardInput {
+                        input:
+                            KeyboardInput {
+                                virtual_keycode: Some(VirtualKeyCode::Escape),
+                                ..
+                            },
+                        ..
+                    },
+                ..
+            } => *control_flow = ControlFlow::Exit,
+            _ => {}
+        }
+    });
+}
+
+#[cfg_attr(target_os = "android", ndk_glue::main(backtrace = "full"))]
+pub fn main() {
+    let event_loop = EventLoop::new();
+    let window = winit::window::WindowBuilder::new()
+        .with_title("Rust GPU - wgpu")
+        .with_inner_size(winit::dpi::LogicalSize::new(1280.0, 720.0))
+        .build(&event_loop)
+        .unwrap();
+
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
I would add `cfg_if` and replace these blocks with that. We're already using it in the `ash` example.

Reply:
Oh that's really nice, I didn't know `cfg_if` existed, thanks for the suggestion!